cmp/mask/ComponentMask.js

```diff
@@ -0,0 +1,54 @@
+/*
+ * This file belongs to Hoist, an application development toolkit
+ * developed by Extremely Heavy Industries (www.xh.io | info@xh.io)
+ *
+ * Copyright © 2018 Extremely Heavy Industries Inc.
+ */
+import {Component} from 'react';
+import {PropTypes as PT} from 'prop-types';
+import {hoistComponent, elemFactory} from 'hoist/core';
+import {frame} from 'hoist/layout';
+import {overlay} from 'hoist/kit/blueprint';
+
+import './Mask.scss';
+
+/**
+ * Simple Mask for Disabled or Inactive components.
+ */
+@hoistComponent()
+export class ComponentMask extends Component {
+
+    BACKGROUND = 'rgba(0,0,0, 0.8)';
+
+    static propTypes = {
+        isDisplayed: PT.bool,
+        text: PT.string
+    };
+
+    render() {
+        const {isDisplayed, text} = this.props;
+        if (!isDisplayed) return null;
+
+        return overlay({
+            cls: 'xh-mask',
+            autoFocus: false,
+            isOpen: true,
+            canEscapeKeyClose: false,
+            backdropProps: {
+                style: {backgroundColor: this.BACKGROUND}
```

Comment:
Spoke to Gil a bit about his changes. We should still be able to set this color via CSS. But should  keep the backdrop as with the addition of Classes.OVERLAY_SCROLL_CONTAINER it will expand to fill it's container and will simply the code thats doing that now.

Will circle back to this a little later on.